seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLFunction.java

```diff
@@ -199,8 +202,29 @@ public Object computeForValue(Expression expression, Object[] inputFields) {
             return ((StringValue) expression).getValue();
         }
         if (expression instanceof Column) {
-            int idx = inputRowType.indexOf(((Column) expression).getColumnName());
-            return inputFields[idx];
+            Column columnExp = (Column) expression;
+            try {
+                String columnName = columnExp.getColumnName();
+                int idx = inputRowType.indexOf(columnName);
+                return inputFields[idx];
+            } catch (IllegalArgumentException e) {
```

Comment:
Can you add a new `indexOf` method in `SeaTunnelRowType` which no IllegalArgumentException exception is thrown. Then use the result of the index to determine whether there is a corresponding fieldname. Because data processing is performed through try catch at now, the performance is particularly bad.